server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -95,8 +100,12 @@ public void assertValid(Username username) throws UsersRepositoryException {
                 throw new InvalidUsernameException("Given Username contains a @domainpart but virtualhosting support is disabled");
             }
         }
+    }
 
-        if (!isLocalPartValid(username)) {
+    private void assertLocalPartValid(Username username) throws InvalidUsernameException {
+        boolean isValid = CharMatcher.anyOf(ILLEGAL_USERNAME_CHARACTERS)
+            .matchesNoneOf(username.getLocalPart());
```

Comment:
Shoudn't knowing if the local part is valid be a responsibility of the Username class?